### PR TITLE
Do a polish/cleanup pass on the docs

### PR DESCRIPTION
**Pull request by spladug**
This tries to streamline, give examples, clarify language, and split out
pages that were getting overly complex.

Here it is rendered: http://baseplate.readthedocs.io/en/docs-polish/

### CODE REVIEW

File: docs/index.rst [hunk 1]
```diff
@@ -3,50 +3,30 @@ Baseplate
...
-applications.  This framework provides monitoring, logging, and tracing out of
-the box.
+Baseplate provides an instrumentation framework that integrates with popular
+application frameworks to provide automatic diagnostics to services.
 
 .. toctree::
    :titlesonly:
 
-   baseplate.core: The guts of the diagnostics framework <baseplate/core>
+   baseplate.core: The skeleton of the instrumentation framework <baseplate/core>
```
Comment (gtaylor) [inline docs/index.rst]: skelton. Come on, Neil.

File: docs/index.rst [hunk 2]
```diff
@@ -3,50 +3,30 @@ Baseplate
...
    baseplate.integration: Integration with application frameworks <baseplate/integration/index>
    baseplate.diagnostics: Diagnostics observers <baseplate/diagnostics/index>
 
 The Library
 -----------
 
-These modules are relatively disconnected and provide various pieces of
-functionality that are commonly needed in production applications. They can
-be used without the framework.
+Baseplate also provides a collection of "extra batteries", independent modules
```
Comment (gtaylor) [inline docs/index.rst]: :no_bell: em dash or colon instead of comma? This reads funky

File: baseplate/context/__init__.py [hunk 3]
```diff
@@ -1,9 +1,8 @@
 """Helpers that integrate common client libraries with baseplate's diagnostics.
 
 This package contains modules which integrate various client libraries with
-Baseplate's diagnostics and tracing facilities. When using these helpers,
-trace information and monitoring are automatically collected and sent on for
-you.
+Baseplate's instrumentation. When using them, trace information is passed on
```
Comment (gtaylor) [inline baseplate/context/__init__.py]: > When using them,

:no_bell: This is only partially ambiguous, but it doesn't read 100% smoothly. May be worth considering different wording.

**Comment by spladug** [comment]
thanks for the feedback :haircut_man:

**Comment by gtaylor** [comment]
:fish:

**Comment by spladug** [comment]
970b287a61eb2b6d7824b67789a03073c61dac6d

**Pull request closed without merging**